src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespacearound/InputWhitespaceAroundLiteralWhen.java

```diff
@@ -0,0 +1,38 @@
+/*
+WhitespaceAround
+tokens = LITERAL_WHEN
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespacearound;
+
+public class InputWhitespaceAroundLiteralWhen {
+
+    void test(Object o) {
+        switch (o) {
+            case Integer i when(i == 0) -> {} // violation, ''when' is not followed by whitespace'
+            // violation below, ''when' is not followed by whitespace'
+            case String s when(s.equals("a")) -> {}
+            // violation below, ''when' is not followed by whitespace'
+            case Point(int x, int y ) when(x>=0 && y >=0) -> {}
```

Comment:
Same as https://github.com/checkstyle/checkstyle/pull/15304#discussion_r1683362869, let's find other tokens to put here.

Reply:
done